app/models/event.rb

```diff
@@ -1,5 +1,6 @@
 # app/models/event.rb
 class Event < ApplicationRecord
+  include EventFrontEnd
```

Comment:
http://www.javaworld.com/article/2074564/learn-java/a-look-at-the-composite-design-pattern.html

I think a presenter could be a good candidate for the composite design pattern, if you need this logic to be available to  the model